src/algorithms/primaldual.jl

```diff
@@ -94,60 +99,94 @@ function Base.iterate(iter::AFBA_iterable, state::AFBA_state=AFBA_state(iter))
+                alpha = beta_l/(par2*nmL)
             end
-            gamma1 = R(1)/(betaQ/2+nmL/alpha)
-            gamma2 = R(0.99)/(betaR/2+nmL*alpha)
-        elseif theta == 1 && mu == 1 # default stepsize for theta=1, mu=1
-            if betaQ > par*nmL && betaR > par*nmL
+            gamma1 = R(1)/(beta_f/2+nmL/alpha)
+            gamma2 = R(0.99)/(beta_l/2+nmL*alpha)
+        elseif theta == 1 && mu == 1 # SPCA
+            if nmL > par2*beta_l # for the case beta_f = 0
                 alpha = R(1)
-            elseif betaQ > par*nmL
-                alpha = 2*nmL/betaQ
-            elseif betaR > par*nmL
-                alpha = betaR/(2*nmL)
+            elseif beta_l > par*beta_f 
+                alpha =  beta_l/(par2*nmL)    
             end
-            gamma1 = R(1)/(betaQ/2+nmL/alpha)
-            gamma2 = R(0.99)/(betaR/2 + gamma1*nmL^2)
-        elseif theta == 0 && mu == 1 # default stepsize for theta=0, mu=1
-            temp=3
-            if betaQ > par*nmL && betaR > temp*par*nmL # denominator for Sigma involves 3α opnorm(L) in this case
-                alpha = R(1)
-            elseif betaQ > par*nmL
-                alpha = 2*nmL/betaQ
-            elseif betaR > temp*par*nmL # denominator for Sigma involves 3α opnorm(L) in this case
-                alpha = betaR/(temp*2*nmL)
+            gamma1 = beta_f > 0 ?  R(1.99)/beta_f : R(1)/(nmL/alpha)
+            gamma2 = R(0.99)/(beta_l/2 + gamma1*nmL^2)
+        elseif theta == 0 && mu == 1 # PPCA
+            temp = R(3)
+            if beta_f == 0
+                nmL *= sqrt(temp)
+                if nmL > par *beta_l
+                    alpha = R(1)
+                else 
+                    alpha = beta_l/(par2*nmL)
+                end
+                gamma1 = R(1)/(beta_f/2+nmL/alpha)
+                gamma2 = R(0.99)/(beta_l/2+nmL*alpha)
+            else  
+                if nmL > par * max(beta_l, beta_f)
+                    alpha = R(1)
+                elseif beta_f > par*beta_l
+                    alpha = par2*nmL/beta_f
+                elseif beta_l > par*beta_f
+                    alpha = beta_l/(par2*nmL)
+                end
+                xi = 1+ 2*nmL/(nmL+alpha*beta_f/2)
+                gamma1 = R(1)/(beta_f/2+nmL/alpha)
+                gamma2 = R(0.99)/(beta_l/2+xi*nmL*alpha)
             end
-            gamma1 = R(1)/(betaQ/2+nmL/alpha)
-            gamma2 = R(0.99)/(betaR/2 + 2*gamma1*nmL^2 + alpha*nmL)
-        elseif mu == 0 # default stepsize for  mu=0
+        elseif mu == 0 # SDCA & PDCA
             temp = theta^2-3*theta+3
-            if betaQ > temp*par*nmL && betaR > par*nmL
-                alpha = R(1)
-            elseif betaR > par*nmL
-                alpha = betaR/(2*nmL)
-            elseif betaQ > temp*par*nmL # denominator for Sigma involves 3α opnorm(L) in this case
-                alpha = 2*nmL*temp/betaQ
+            if beta_l == R(0)
+                nmL *= sqrt(temp)
+                 if nmL > par *beta_f
```

Comment:
Still off by one?

Reply:
It seems I have used space rather than tab at some point, that is causing this issue....I thought I removed all those...